### ISSUE TITLE
Support regular expression operator

### ISSUE THREAD

**Issue by mads-hartmann** [tree-sitter/tree-sitter-bash#5]
Currently the parser will be unhappy with the following expression

```bash
[[ "35d8b" =~ ^[0-9a-fA-F]{5}$ ]] && echo "It's a hash"
```
My initial idea was to simple create `$.regex` rule and add that to `$._expression` but given that there aren't any marker characters for regular expressions (like `/abc/` in [javascript](https://github.com/tree-sitter/tree-sitter-javascript/blob/master/grammar.js#L747)) in Bash I think we have to go another route. I'm not sure what the best approach is at the moment so I'll just leave the issue open 😄

**Comment by maxbrunsfeld** [comment]
Thanks for the report! I'm just reading about this `=~` operator in the [bash scripting guide](http://tldp.org/LDP/abs/html/bashver3.html#BASH3REF).

> NOTE: As of version 3.2 of Bash, expression to match no longer quoted.

It looks like newer versions of bash have special rules for tokenizing the regular expression after the `=~` operator. So we might want a special `regex` token after all, which would only be allowed to appear after `=~`.

**Comment by maxbrunsfeld** [comment]
Actually, nevermind; this is not specific to the `=~` operator; it looks like we just need to loosen our definition of what constitutes a `word` (bare value passed as an argument). Because it seems like all of these are valid:

```sh
echo ^[0-9a-fA-F]{5}$
echo {}
echo $
```

**Comment by mads-hartmann** [comment]
Oh I had no idea. Sounds like to right solution 👍

**Comment by maxbrunsfeld** [comment]
Actually there does seem to special parsing for regular expressions after the `=~` sign.

For example, this parses fine:

```sh
[[ $CMD =~ (^|;)update_terminal_cwd($|;) ]]
```

But this does not:

```sh
echo (^|;)update_terminal_cwd($|;)
```

**Comment by maxbrunsfeld** [comment]
It seems like in certain places (maybe just after `=~`, we need special tokenization of words where we allow any non-whitespace character.

**Comment by maxbrunsfeld** [comment]
This also seems valid:

```sh
echo ${CMD/(^|;)update_terminal_cwd($|;)}
```

**Comment by mads-hartmann** [comment]
@maxbrunsfeld Hehe, seems like writing a grammar for a language will expose some of the more esoteric features ;) Nice catch

**Comment by maxbrunsfeld** [comment]
Fixed via c7e42b8e9624d193be7687d8fb8ce33a946ba336 and 079102759679e8b96b699d4cf893272c271bd04e.
